pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -306,14 +308,32 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 			return nil, httperror.WrapAPIError(err, httperror.ServerError, "Error processing clusterTemplate answers")
 		}
 		keyParts := strings.Split(question.Variable, ".")
-		values.PutValue(dataFromTemplate, val, keyParts...)
+		if strings.HasPrefix(question.Variable, registryConfigPath) {
+			key := keyParts[len(keyParts)-1]
```

Comment:
Is there any validation on the request before this stage? If not there is a possibility for an index error here.

Reply:
There is no validation on this specific field/section before this stage. But since the field comes from `clusterTemplateRevision.Spec.Questions`, unless users are manipulating the questions when creating the clusterTemplateRevision, there should not be an index error. 
However, I will add a check here to avoid the potential error.

Reply:
Regardless of the value of `question.Variable`, it is impossible to have an index error here. From the documentation: `If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s.` This means that even if `question.Variable` is empty, then `keyParts` will have at least length 1.

Reply:
closed this one since it is not an issue.